coalib/misc/i18n.py

```diff
@@ -8,40 +8,47 @@
 COALA_DOMAIN = 'coala'
 
 
+def file_needs_update(source, generated):
+    """
+    Checks if the generated file is nonexistent or older than the source file
+    and to be regenerated.
+    """
+    if os.path.exists(generated):
+        src_mtime = os.stat(source)[8]
+        gen_mtime = os.stat(generated)[8]
+        return src_mtime > gen_mtime
+    else:
+        return True
+
+
 def compile_translations():
     translations = []
     trans_install_dir_prefix = os.path.join(sys.prefix, "share", "locale")
     for (path, dirnames, filenames) in os.walk("locale"):
-        for filename in filenames:
-            if filename.endswith(".po"):
-                lang = filename[:-3]
-                src = os.path.join(path, filename)
-                dest_path = os.path.join("build",
-                                         "locale",
-                                         lang,
-                                         "LC_MESSAGES")
-                dest = os.path.join(dest_path, COALA_DOMAIN + ".mo")
-                install_dir = os.path.join(trans_install_dir_prefix,
-                                           lang,
-                                           "LC_MESSAGES")
-
-                if not os.path.exists(dest_path):
-                    os.makedirs(dest_path)
-                elif os.path.exists(dest):
-                    src_mtime = os.stat(src)[8]
-                    dest_mtime = os.stat(dest)[8]
-                    if src_mtime <= dest_mtime:
-                        translations.append((install_dir, [dest]))
-                        continue
-
-                try:
+        for filename in filter(lambda name: name.endswith(".po"), filenames):
+            lang = filename[:-3]
+            src = os.path.join(path, filename)
+            dest_path = os.path.join("build",
+                                     "locale",
+                                     lang,
+                                     "LC_MESSAGES")
+            os.makedirs(dest_path, exist_ok=True)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Unexpected keyword argument 'exist_ok' in function call](https://www.codacy.com/app/lasse/coala/file/751657166/issues/source?bid=2303100&fileBranchId=2447283#l35)